File: eMovieFinder/eMovieFinder.Database/Entities/Director.cs
using eMovieFinder.Model.Utilities;
using System.ComponentModel.DataAnnotations;

namespace eMovieFinder.Database.Entities
{
    public class Director : TimeStampObject
    {
        [Key]
        public int Id { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public DateTime BirthDate { get; set; }
    }
}

File: eMovieFinder-UI/emovie_finder_mobile/lib/src/models/entities/director.dart
import 'package:json_annotation/json_annotation.dart';

import 'package:emovie_finder_mobile/src/models/utilities/time_stamp_object.dart';

part 'director.g.dart';

@JsonSerializable()
class Director extends TimeStampObject {
  final int? id;
  final String? firstName;
  final String? lastName;
  final DateTime? birthDate;

  Director({
    super.creationDate,
    super.modifiedDate,
    super.createdById,
    super.modifiedById,
    super.formattedCreationDate,
    super.formattedModifiedDate,
    this.id,
    this.firstName,
    this.lastName,
    this.birthDate
  });

  factory Director.fromJson(Map<String, dynamic> json) => _$DirectorFromJson(json);
  Map<String, dynamic> toJson() => _$DirectorToJson(this);
}